src/components/chat/chat-interface.tsx

```diff
@@ -0,0 +1,105 @@
+"use client";
+
+import { useChat } from "@ai-sdk/react";
+import ReactMarkdown from "react-markdown";
+import type { Components } from "react-markdown";
+
+export default function ChatInterface() {
+  const { messages, input, handleInputChange, handleSubmit, isLoading, error } =
+    useChat({
+      api: "/api/chat",
+    });
+
+  const components: Components = {
```

Comment:
Make sure to create components outside the ChatInterface component, you can keep it in the same file. This would improve performance

Reply:
Fixed